ledger/internal/prefetcher/prefetcher.go

```diff
@@ -283,21 +287,60 @@ func (p *accountPrefetcher) prefetch(ctx context.Context) {
 
 	// iterate over the transaction groups and add all their account addresses to the list
 	queue := &tasksQueue
-	for i := range p.groups {
+	for i := range p.txnGroups {
 		task := &groupsReady[i]
-		for j := range p.groups[i] {
-			stxn := &p.groups[i][j]
+		for j := range p.txnGroups[i] {
+			stxn := &p.txnGroups[i][j]
 			switch stxn.Txn.Type {
 			case protocol.PaymentTx:
 				loadAccountsAddAccountTask(&stxn.Txn.Receiver, task, accountTasks, queue)
 				loadAccountsAddAccountTask(&stxn.Txn.CloseRemainderTo, task, accountTasks, queue)
 			case protocol.AssetConfigTx:
+				loadAccountsAddResourceTask(nil, basics.CreatableIndex(stxn.Txn.ConfigAsset), basics.AssetCreatable, task, resourceTasks, queue)
```

Comment:
This takes a cache miss on initial acfg creation when Txn.ConfigAsset == 0. We _could_ know that id based on block header txn count and the position within the payset. Also probably not a big deal, pretty rare operation, rare cache miss.

Reply:
There are two different things here worth thinking about, I'm not sure which one you mean.  First, in an acfg with id=0, we should not try to prefetch the 0 ASA, since we know it doesn't exist.  It looks like we handle that case in `loadAccountsAddResource`.

And then there's the question of whether block validation is slowed down because we didn't load the thing that will eventually be created.  I don't think so. At evaluation time, acfg create should not be looking up the newly created id. (This brings up a silly optimization we should do, though it presumably does not happen in "good" txns - short-circuit the db lookup for creatables that can't exist because the id is too high.)

Also, need to be careful now about predicting the id based on position in payset. Inner txns count, so id is not just the block start id + payset index.